sklearn/ensemble/_hist_gradient_boosting/splitting.pyx

```diff
@@ -465,19 +522,106 @@ cdef class Splitter:
                                negative_loss_current_node,
                                self.l2_regularization)
 
-            if gain > best_split.gain and gain > self.min_gain_to_split:
-                best_split.gain = gain
-                best_split.feature_idx = feature_idx
-                best_split.bin_idx = bin_idx
-                best_split.sum_gradient_left = sum_gradient_left
-                best_split.sum_gradient_right = sum_gradient_right
-                best_split.sum_hessian_left = sum_hessian_left
-                best_split.sum_hessian_right = sum_hessian_right
-                best_split.n_samples_left = n_samples_left
-                best_split.n_samples_right = n_samples_right
+            if gain > split_info.gain and gain > self.min_gain_to_split:
+                split_info.gain = gain
+                split_info.feature_idx = feature_idx
+                split_info.bin_idx = bin_idx
+                # we scan from left to right so missing values go to the right
+                split_info.missing_go_to_left = False
+                split_info.sum_gradient_left = sum_gradient_left
+                split_info.sum_gradient_right = sum_gradient_right
+                split_info.sum_hessian_left = sum_hessian_left
+                split_info.sum_hessian_right = sum_hessian_right
+                split_info.n_samples_left = n_samples_left
+                split_info.n_samples_right = n_samples_right
+
+    cdef void _find_best_bin_to_split_right_to_left(
```

Comment:
This function could not be factorized with the previous one in which depending of a bool we could go left to right or right to left?

Reply:
There are some subtle differences between both functions.

Factorizing them would make the code full of if/else, and much more complicated. It's not worth it.

Reply:
OK good with me.